File: frontend/types/document.ts
export interface DocumentSummary {
  document_id: string;
  original_filename: string;
  content_type: string;
  size_bytes: number;
  uploaded_at: string;
  chunk_count: number;
  status: string;
}

export interface DocumentListResponse {
  documents: DocumentSummary[];
}

File: backend/app/schemas/document.py
from pydantic import BaseModel


class DocumentSummary(BaseModel):
    document_id: str
    original_filename: str
    content_type: str
    size_bytes: int
    uploaded_at: str
    chunk_count: int
    status: str


class DocumentListResponse(BaseModel):
    documents: list[DocumentSummary]